airbyte-integrations/connectors/destination-postgres/src/main/kotlin/io/airbyte/integrations/destination/postgres/sql/PostgresDirectLoadSqlGenerator.kt

```diff
@@ -84,56 +71,63 @@ class PostgresDirectLoadSqlGenerator(
         tableName: TableName,
         columnNameMapping: ColumnNameMapping
     ): String {
-        val primaryKeyIndexStatement = getPrimaryKeysColumnNames(stream, columnNameMapping)
-            .takeIf {  it.isNotEmpty() }
-            ?.let {
-                "CREATE INDEX ${getPrimaryKeyIndexName(tableName)} ON ${getFullyQualifiedName(tableName)} (${it.joinToString(", ")});"
-            } ?: ""
-
-        val cursorIndexStatement = getCursorColumnName(stream, columnNameMapping)?.let { cursorColumnName ->
-            "CREATE INDEX ${getCursorIndexName(tableName)} ON ${getFullyQualifiedName(tableName)} ($cursorColumnName);"
-        } ?: ""
-
+        val primaryKeyColumnNames = postgresColumnUtils.getPrimaryKeysColumnNames(stream, columnNameMapping)
+        val primaryKeyIndexStatement = createPrimaryKeyIndexStatement(primaryKeyColumnNames, tableName)
+        val cursorColumnName = postgresColumnUtils.getCursorColumnName(stream, columnNameMapping)
+        val cursorIndexStatement = createCursorIndexStatement(cursorColumnName, tableName)
         val extractedAtIndexStatement = "CREATE INDEX ON ${getFullyQualifiedName(tableName)} ($EXTRACTED_AT_COLUMN_NAME);"
 
         return """
             $primaryKeyIndexStatement
             $cursorIndexStatement
             $extractedAtIndexStatement
-        """.trimIndent()
+        """
     }
 
-    private fun getPrimaryKeysColumnNames(stream: DestinationStream, columnNameMapping: ColumnNameMapping): List<String> {
-        return when (stream.importType) {
-            is Dedupe -> getPrimaryKeysColumnNames(stream.importType as Dedupe, columnNameMapping)
-            else -> listOf()
-        }
+    internal fun recreatePrimaryKeyIndex(
+        primaryKeyColumnNames: List<String>,
+        tableName: TableName
+    ): String {
+        val dropPrimaryKeyIndexStatement = dropIndex(getPrimaryKeyIndexName(tableName))
+        val primaryKeyIndexStatement = createPrimaryKeyIndexStatement(primaryKeyColumnNames, tableName)
+
+        return """
+            $dropPrimaryKeyIndexStatement
+            $primaryKeyIndexStatement
+        """
     }
 
-    private fun getPrimaryKeysColumnNames(
-        importType: Dedupe,
-        columnNameMapping: ColumnNameMapping
-    ): List<String> {
-        return importType.primaryKey.map { fieldPath ->
-            val primaryKeyColumnName = fieldPath.first() //only at the root level for Postgres
-            getTargetColumnName(primaryKeyColumnName, columnNameMapping )
-        }.toList()
+    private fun createPrimaryKeyIndexStatement(
```

Comment:
moved to a separate function to use it when creating the table and when recreating the index